admin/bin/qserv-configure.py

```diff
@@ -1,311 +1,381 @@
-                        default=False,
-                        help="forcing removal of existing execution data"
-                        )
-
-    # run dir, all mutable data related to a qserv running instance are
-    # located here
-    parser.add_argument("-R", "--qserv-run-dir", dest="qserv_run_dir",
-                        default=default_qserv_run_dir,
-                        help="full path to qserv_run_dir"
-                        )
-
-    # meta-configuration file whose parameters will be dispatched in Qserv
-    # services configuration files
-    args = parser.parse_args()
-    default_meta_config_file = os.path.join(
-        args.qserv_run_dir, "qserv-meta.conf")
-    parser.add_argument("-m", "--metaconfig", dest="meta_config_file",
-                        default=default_meta_config_file,
-                        help="full path to Qserv meta-configuration file"
-                        )
-
-    args = parser.parse_args()
-
-    if args.all:
-        args.step_list = configure.STEP_LIST
-    elif args.step_list is None:
-        args.step_list = configure.STEP_RUN_LIST
-
-    args.verbose_level = verbose_dict[args.verbose_str]
-
-    return args
-
-
-def main():
-
-    args = parseArgs()
-
-    logging.basicConfig(
-        format='%(levelname)s: %(message)s', level=args.verbose_level)
-
-    logging.info("Qserv configuration tool\n" +
-                 "======================================================================="
-                 )
-
-    qserv_dir = os.path.abspath(
-        os.path.join(
-            os.path.dirname(os.path.realpath(__file__)),
-            "..")
-    )
-
-    if configure.PREPARE in args.step_list:
-
-        if os.path.exists(args.qserv_run_dir):
-            if args.force or configure.user_yes_no_query(
-                "WARNING : Do you want to erase all configuration" +
-                " data in {0} ?".format(args.qserv_run_dir)
-            ):
-                shutil.rmtree(args.qserv_run_dir)
-            else:
-                logging.info(
-                    "Stopping Qserv configuration, please specify an other configuration directory")
-                sys.exit(1)
+        parser.add_argument('-v', '--verbose', dest='verbose', default=[],
+                            action='append_const',
+                            const=None,
+                            help='More verbose output, can use several times.')
+        parser = lsst.qserv.admin.logger.add_logfile_opt(parser)
+
+        parser.add_argument("-a", "--all", dest="all", action='store_true',
+                            default=False,
+                            help='''clean execution directory and then run all configuration steps'''
+                            )
+        # Defining option of each configuration step
+        for step_name in configure.STEP_LIST:
+            parser.add_argument(
+                "-{0}".format(configure.STEP_ABBR[step_name]),
```

Comment:
Is `format()` even needed here, `"-" + configure.STEP_ABBR[step_name]` is shorter and more efficient.